docs/tutorials/upgrade.md

```diff
@@ -1,17 +1,84 @@
-the upgrade guides from the v2.5 releases}, then return to this guide.
+ - {@link https://v2-4-7-dot-shaka-player-demo.appspot.com/docs/api/tutorial-upgrade.html Upgrade to v2.4}
+ - {@link https://v2-5-23-dot-shaka-player-demo.appspot.com/docs/api/tutorial-upgrade-v2-4.html Upgrade v2.4 => v2.5}
+ - {@link https://v3-0-15-dot-shaka-player-demo.appspot.com/docs/api/tutorial-upgrade-v2-5.html Upgrade v2.5 => v3.0}
 
-From these older release branches, you will have to use our upgrade guides in
-phases.  This is necessary to keep our guides maintainable.  Thanks for your
-understanding!
+Since v3.0, Shaka Player has been following semantic versioning.  (The
+IE11 deprecation announced before v3.0 happened in v3.1, which technically
+breaks semantic versioning guarantees.  It is the only intentional exception.)
 
-If you are upgrading from **v2.5**, please see {@tutorial upgrade-v2-5}.
+Upgrading from any v3 release to a newer v3 release should be backward
+compatible.  The same is true of all major version numbers (v4 => v4, etc).
 
-If you are upgrading from another **v3** release, please see {@tutorial upgrade-v3}.
+Here is a summary of breaking changes that might require upgrades to your
+application:
+
+
+## v3.1
+
+  - New dependencies:
+    - TextDecoder/TextEncoder platform support or polyfill required (affects
+      Xbox One, but not evergreen browsers); we suggest the polyfill
+      [https://github.com/anonyco/FastestSmallestTextEncoderDecoder](fastestsmallesttextencoderdecoder/EncoderDecoderTogether.min.js)
+
+  - Support removed:
+    - IE11 support removed
+
+
+## v4.0
+
+  - Support removed:
+    - Older TVs and set-top boxes that do not support MediaSource sequence mode
+      can no longer play HLS content (since we now use sequence mode for that)
+
+  - Configuration changes:
+    - `manifest.dash.defaultPresentationDelay` has been replaced by
+      `manifest.defaultPresentationDelay` (deprecated in v3.0.0)
+    - Configuration of factories should be plain factory functions, not
+      constructors; these will not be invoked with `new` (deprecated in v3.1.0)
+
+  - Player API changes:
+    - `shaka.Player.prototype.addTextTrack()` has been replaced by
+      `addTextTrackAsync()`, which returns a `Promise` (deprecated in v3.1.0)
+
+  - UI API changes:
+    - `shaka.ui.TrackLabelFormat` has been renamed to
+      `shaka.ui.Overlay.TrackLabelFormat` (deprecated in v3.1.0)
+    - `shaka.ui.FailReasonCode` has been renamed to
+      `shaka.ui.Overlay.FailReasonCode` (deprecated in v3.1.0)
+
+  - Offline API changes:
+    - `shaka.offline.Storage.prototype.store()` returns `AbortableOperation`
+      instead of `Promise`; callers should change `.then()` to
+      `.promise.then()` and `await rv` to `await rv.promise` (deprecated in
+      v3.0.0)
+    - `shaka.offline.Storage.prototype.getStoreInProgress()` has been removed;
+      concurrent operations are supported, so callers don't need to check this
```

Comment:
Should probably specify concurrent operations are "now" supported, to make it clear that that was a change. As-is, it makes it sound like `getStoreInProgress` was an unnecessary feature we are removing, or something.

Reply:
Done, thanks!